build/opt.go

```diff
@@ -353,23 +353,34 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		so.Internal = true
 	}
 
-	return &so, releaseF, nil
+	return &so, releaseF, dockerfileMapping, nil
 }
 
-func loadInputs(ctx context.Context, d *driver.DriverHandle, inp Inputs, pw progress.Writer, target *client.SolveOpt) (func(), error) {
+type DockerfileMapping struct {
```

Comment:
Updated to do it this way - I still ended up returning a `build.Inputs` because the top-level calls we need to return this information from use `controllerapi.BuildOptions` as an arg, and I didn't think updating that was the right call (because adding the info to GRPC-land seems like the wrong approach when we don't need to communicate this information across client/server boundary).